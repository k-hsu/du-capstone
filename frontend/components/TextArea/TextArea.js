import React, { forwardRef } from "react";
import styled from "@emotion/styled";
import Text from "../Text/Text";
import { color, fontWeight, spacing } from "../../theme";

const StyledLabel = styled.label`
  display: flex;
  flex-direction: column;
`;

const StyledTextArea = styled.textarea`
  border-radius: ${spacing["0.5"]};
  padding: ${spacing["0.5"]} ${spacing["0.75"]};
`;

const TextArea = forwardRef(
  ({ id, labelText, errorMessage, ...props }, ref) => {
    return (
      <StyledLabel htmlFor={id}>
        <Text fontWeight={fontWeight.bold}>{labelText}</Text>
        <StyledTextArea id={id} ref={ref} {...props} />
        {errorMessage && <Text color={color.red}>{errorMessage}</Text>}
      </StyledLabel>
    );
  }
);
TextArea.displayName = "TextArea";

export default TextArea;
