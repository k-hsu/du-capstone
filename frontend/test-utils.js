import { render } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import { renderHook } from "@testing-library/react-hooks";

function setup(jsx) {
  return {
    user: userEvent.setup(),
    ...render(jsx),
  };
}

export * from "@testing-library/react";
export { renderHook, setup };
