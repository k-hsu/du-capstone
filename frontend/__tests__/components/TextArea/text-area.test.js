import React, { useRef } from "react";
import { render, screen, userEvent } from "../../../test-utils";
import TextArea from "../../../components/TextArea/TextArea";

describe("Text Area", () => {
  const RefComponent = (props) => {
    const ref = useRef();
    return <TextArea ref={ref} {...props} />;
  };
  it("should render text area component", () => {
    render(<RefComponent labelText="description" />);
    expect(screen.getByText("description")).toBeInTheDocument();
    expect(
      screen.getByRole("textbox", { name: "description" })
    ).toBeInTheDocument();
  });
  it("should show values as typed by user", () => {
    render(<RefComponent labelText="description" />);

    userEvent.type(
      screen.getByRole("textbox", { name: "description" }),
      "The #2 pop song of the year 1982"
    );
    expect(screen.getByRole("textbox", { name: "description" })).toHaveValue(
      "The #2 pop song of the year 1982"
    );
  });
});
