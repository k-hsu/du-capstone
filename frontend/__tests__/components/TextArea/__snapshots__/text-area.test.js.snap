// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Text Area should render text area component styles 1`] = `
<label
  className="css-1iyoj2o"
>
  <p
    className="css-1v2p52x"
    fontFamily="IBM Plex Sans"
    fontWeight="700"
  >
    prompt
  </p>
  <textarea
    className="css-mn8zue"
  />
</label>
`;
